test.l

```diff
@@ -1,31 +1,34 @@
 #!/usr/bin/env bin/lumen
 
+(when-compiling
+  (define-macro test (x msg)
+    `(if (not ,x)
+         (do (set failed (+ failed 1))
+             (return ,msg))
+       (inc passed)))
+
+  (define-macro test= (a b)
+    (let-unique (x y)
+      `(let (,x ,a ,y ,b)
+         (test (equal? ,x ,y)
+               (cat "failed: expected " (str ,x) ", was " (str ,y))))))
+
+  (define-macro define-test (name rest: body)
+    `(add tests (list ',name (fn () ,@body))))
+
+  nil)
```

Comment:
After sleeping on it, I think this PR is a good idea. Here are my reasons:

1. Both Lumen and Motor load macros separately. They don't define macros in the same compilation unit.

2. Thanks to `when-compiling`, we have a way of defining macros in the same compilation unit as they're used (aka "local macros").

3. We can easily add an `:eval` option to evaluate a macro immediately at compile time.

4. I think the benefits of using local functions in a macro outweigh the small tradeoff in simplicity. This is a judgement call, and you should have the final word on it.